packages/server/src/middleware/http-logging-middleware.ts

```diff
@@ -0,0 +1,27 @@
+import type { RequestHandler } from 'express';
+import { isGeneratorFunction } from '../guards/guards';
+import { getFormattedDate } from './get-formatted-date';
+import { log } from './template';
+
+export interface Logger {
+  (log: string): void;
+}
+
+export function createLoggingMiddleware(...loggers: Logger[]): RequestHandler {
+  return function * (req, res) {
+    let date = getFormattedDate();
+    let method = req.method;
+    let status = res.statusCode;
+
+    let logMessage = log({ date, method, req, status });
+
+    for(let log of loggers) {
+      if(isGeneratorFunction(log)) {
+        yield log(logMessage);
+        continue;
+      }
+
+      log(logMessage);
```

Comment:
Let's just make all loggers operations. We ran afoul of that before when trying to do that with streams so that you could use them with normal fat-arrow functions, and in the ended it turned out being better to just have everything be normalized. A generator will work even if there are no yields in it.

Also, I think we should pass in the `request and response` to the logger so that if the string isn't enough it can do its own thing.